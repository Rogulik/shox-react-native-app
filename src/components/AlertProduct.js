import React, { useRef } from "react";
import { Animated, StyleSheet, Text } from "react-native";

const AlertProduct = (props) => {
  const { text, propsContainerStyles = {}, fadeAnim } = props;

  return (
    <Animated.View
      style={[
        styles.alertContainer,
        { opacity: fadeAnim },
        propsContainerStyles,
      ]}
    >
      <Text style={styles.alertMessage}>{text}</Text>
    </Animated.View>
  );
};

export default AlertProduct;

const styles = StyleSheet.create({
  alertContainer: {
    height: 50,
    backgroundColor: "green",
    position: "absolute",
    bottom: 0,
    left: 0,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  alertMessage: {
    color: "white",
  },
});
