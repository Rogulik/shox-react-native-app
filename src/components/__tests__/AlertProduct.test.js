import React from "react";
import { create } from "react-test-renderer";
import AlertProduct from "../AlertProduct";

describe("AlertProduct", () => {
  it("should render alert", () => {
    const tree = create(<AlertProduct />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render alert with red background color", () => {
    const tree = create(
      <AlertProduct propsContainerStyles={{ backgroundColor: "red" }} />
    ).toJSON();
    expect(tree.props.style.backgroundColor).toEqual("red");
  });
  it("It should render with the text test", () => {
    const tree = create(<AlertProduct text="test" />).root;
    expect(tree.findByType("Text").props.children).toBe("test");
  });
});
