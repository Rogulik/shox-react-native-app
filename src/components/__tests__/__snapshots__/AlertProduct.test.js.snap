// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertProduct should render alert 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "bottom": 0,
      "height": 50,
      "justifyContent": "center",
      "left": 0,
      "opacity": undefined,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  />
</View>
`;
